stdsimd-verify/src/lib.rs

```diff
@@ -0,0 +1,248 @@
+                "i64x2" => my_quote! { &I64x2 },
+                "i64x4" => my_quote! { &I64x4 },
+                "i8" => my_quote! { &I8 },
+                "i8x16" => my_quote! { &I8x16 },
+                "i8x32" => my_quote! { &I8x32 },
+                "i8x8" => my_quote! { &I8x8 },
+                "u16x4" => my_quote! { &U16x4 },
+                "u16x8" => my_quote! { &U16x8 },
+                "u32" => my_quote! { &U32 },
+                "u32x2" => my_quote! { &U32x2 },
+                "u32x4" => my_quote! { &U32x4 },
+                "u32x8" => my_quote! { &U32x8 },
+                "u64" => my_quote! { &U64 },
+                "u64x2" => my_quote! { &U64x2 },
+                "u64x4" => my_quote! { &U64x4 },
+                "u8" => my_quote! { &U8 },
+                "u16" => my_quote! { &U16 },
+                "u8x16" => my_quote! { &U8x16 },
+                "u8x32" => my_quote! { &U8x32 },
+                "u16x16" => my_quote! { &U16x16 },
+                "u8x8" => my_quote! { &U8x8 },
+                s => panic!("unspported type: {}", s),
+            }
+        }
+        syn::Type::Ptr(syn::TypePtr { ref ty, .. }) |
+        syn::Type::Reference(syn::TypeReference { ref ty, .. }) => {
+            let tokens = to_type(&ty.ty);
+            my_quote! { &Type::Ptr(#tokens) }
+        }
+        syn::Type::Slice(_) => panic!("unsupported slice"),
+        syn::Type::Array(_) => panic!("unsupported array"),
+        syn::Type::Tup(_) => panic!("unsupported tup"),
+        _ => panic!("unsupported type"),
+    }
+}
+
+fn extract_path_ident(path: &syn::Path) -> syn::Ident {
+    if path.leading_colon.is_some() {
+        panic!("unsupported leading colon in path")
+    }
+    if path.segments.len() != 1 {
+        panic!("unsupported path that needs name resolution")
+    }
+    match path.segments.get(0).item().arguments {
+        syn::PathArguments::None => {}
+        _ => panic!("unsupported path that has path arguments")
+    }
+    path.segments.get(0).item().ident
+}
+
+fn walk(root: &Path, files: &mut Vec<syn::File>) {
+    for file in root.read_dir().unwrap() {
+        let file = file.unwrap();
+        if file.file_type().unwrap().is_dir() {
+            walk(&file.path(), files);
+            continue
+        }
+        let path = file.path();
+        if path.extension().and_then(|s| s.to_str()) != Some("rs") {
+            continue
+        }
+
+        let mut contents = String::new();
+        File::open(&path).unwrap().read_to_string(&mut contents).unwrap();
+
+        files.push(syn::parse_str::<syn::File>(&contents).expect("failed to parse"));
+    }
+}
+
+fn find_instrs(attrs: &[syn::Attribute]) -> Vec<syn::Ident> {
+    attrs.iter()
+        .filter_map(|a| a.meta_item())
+        .filter_map(|a| {
+            match a {
+                syn::MetaItem::List(i) => {
+                    if i.ident == "cfg_attr" {
+                        Some(i.nested.into_vec())
```

Comment:
Would it be possible to map this to `None` if the `cfg_attr` is disabled. For example, `#[cfg_attr(test, ...` is always enabled on test builds, but `#[cfg_attr(all(test, not(windows)), ...]` should fail on windows builds, so we would like the windows CI to detect when an intrinsic is missing the `assert_instr` check for windows builds.

Reply:
Not easily, no.

Reply:
I see, well we can always add this in the future.